.circleci/workflows.yml

```diff
@@ -732,7 +732,7 @@ jobs:
   artifact-deployment:
     executor: gcp-cli/google
     steps:
-      - run:
+      - run: &oidc-setup
```

Comment:
While this anchor makes sense, the final step https://github.com/mozilla/bigquery-etl/blob/main/.circleci/workflows.yml#L743 is going to have a different value for each distinct SA we're assuming. So we probably want to parameterize this in some fashion because https://app.circleci.com/settings/project/github/mozilla/bigquery-etl/environment-variables?return-to=https%3A%2F%2Fapp.circleci.com%2Fpipelines%2Fgithub%2Fmozilla%2Fbigquery-etl currently only has `GCP_SERVICE_ACCOUNT_EMAIL` set to the airflow DAG triggering SA. This workload uses the SA defined in https://github.com/mozilla-services/cloudops-infra/blob/master/projects/airflow/tf/prod/envs/prod/resources/main.tf#L151-L173

Reply:
Ok so what I'd normally do here is just add a step to overwrite the ennvar afterwards
Because the yaml node here is just a set of strings, theres not a way to use the override `<<` operator either since we'd have to redefine the entire `command` node :|

Reply:
So I'd just handle it like this:
https://github.com/mozilla/bigquery-etl/pull/5898/commits/e3ee174e482c619c53df49f2529c3a098f36c48a#diff-7bf091450b804ea25dc3d11ac28c50202d19178f7b53f54a02ad59c946afa6a7R860

within the confines of yaml and bash I think just setting the new value over the top is clean enough? Unless we want to make a custom orb for this that accepts the SA as a parameter. I'm not opposed to that if theres other OIDC needs out here we can solve with it but the effort-to-outcome ratio seems a bit steep

Reply:
resolved in DMs